tasks/eth/006-MCP-L1/SignFromJsonWithAssertions.s.sol

```diff
@@ -0,0 +1,353 @@
+
+        require(ProtocolVersion.unwrap(protocolVersionsToCheck.recommended()) == recommendedProtocolVersion, "7000");
+    }
+
+    /// @notice Asserts that the SuperchainConfig is setup correctly
+    function checkSuperchainConfig() internal view {
+        console.log("Running chain assertions on the SuperchainConfig");
+
+        require(proxies.SuperchainConfig.code.length != 0, "7100");
+
+        SuperchainConfig superchainConfigToCheck = SuperchainConfig(proxies.SuperchainConfig);
+
+        require(superchainConfigToCheck.guardian() == superchainConfigGuardian, "7200");
+        require(superchainConfigToCheck.guardian().code.length != 0, "7201");
+
+        require(superchainConfigToCheck.paused() == false, "7300");
+    }
+
+    /// @notice Checks the correctness of the deployment
+    function _postCheck() internal view override {
+        console.log("Running post-deploy assertions");
+
+        checkSystemConfig();
+        checkL1CrossDomainMessenger();
+        checkL1StandardBridge();
+        checkL2OutputOracle();
+        checkOptimismMintableERC20Factory();
+        checkL1ERC721Bridge();
+        checkOptimismPortal();
+        checkProtocolVersions();
+        checkSuperchainConfig();
+    }
+
+    /// @notice Reads the contract addresses from lib/superchain-registry/superchain/extra/addresses/mainnet/op.json
+    function _getContractSet() internal returns (Types.ContractSet memory _proxies) {
+        string memory addressesJson;
+
+        // Read addresses json
+        try vm.readFile(
+            string.concat(vm.projectRoot(), "/lib/superchain-registry/superchain/extra/addresses/mainnet/op.json")
+        ) returns (string memory data) {
+            addressesJson = data;
+        } catch {
+            revert("Failed to read lib/superchain-registry/superchain/extra/addresses/mainnet/op.json");
+        }
+
+        _proxies.L1CrossDomainMessenger = stdJson.readAddress(addressesJson, "$.L1CrossDomainMessengerProxy");
+        _proxies.L1StandardBridge = stdJson.readAddress(addressesJson, "$.L1StandardBridgeProxy");
+        _proxies.L2OutputOracle = stdJson.readAddress(addressesJson, "$.L2OutputOracleProxy");
+        _proxies.OptimismMintableERC20Factory =
+            stdJson.readAddress(addressesJson, "$.OptimismMintableERC20FactoryProxy");
+        _proxies.OptimismPortal = stdJson.readAddress(addressesJson, "$.OptimismPortalProxy");
+        _proxies.OptimismPortal2 = stdJson.readAddress(addressesJson, "$.OptimismPortalProxy");
+        _proxies.SystemConfig = stdJson.readAddress(addressesJson, "$.SystemConfigProxy");
+        _proxies.L1ERC721Bridge = stdJson.readAddress(addressesJson, "$.L1ERC721BridgeProxy");
+
+        // Get go path
+        string[] memory inputs = new string[](3);
+        inputs[0] = "go";
+        inputs[1] = "env";
+        inputs[2] = "GOPATH";
+        string memory goPath = string(vm.ffi(inputs));
+
```

Comment:
Hmm we should't couple this to the go path, it seems like you should add the go path to your `$PATH` in your `.bashrc`